src/v/kafka/server/group_stm.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+#include "cluster/fwd.h"
+#include "cluster/partition.h"
+#include "kafka/protocol/fwd.h"
+#include "kafka/server/logger.h"
+#include "kafka/server/member.h"
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "model/record.h"
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/util/log.hh>
+
+#include <absl/container/node_hash_map.h>
+#include <absl/container/node_hash_set.h>
+
+#include <iosfwd>
+#include <optional>
+#include <vector>
+
+namespace kafka {
+
+/**
+ * the value type of a group metadata log record.
+ */
+struct group_log_group_metadata {
+    kafka::protocol_type protocol_type;
+    kafka::generation_id generation;
+    std::optional<kafka::protocol_name> protocol;
+    std::optional<kafka::member_id> leader;
+    int32_t state_timestamp;
```

Comment:
this is the only member here without default initialization. should we set to -1

Reply:
I've just moved this type from one file to another, changing its defaults looks like a more disruptive change I prefer not to do it, at least not within this PR